crates/wit-parser/src/decoding.rs

```diff
@@ -46,8 +47,8 @@ enum WitEncodingVersion {
 impl ComponentInfo {
     /// Creates a new component info by parsing the given WebAssembly component bytes.
 
-    fn from_reader(mut reader: impl Read) -> Result<Self> {
-        let mut validator = Validator::new();
+    fn from_reader(mut reader: impl Read, features: WasmFeatures) -> Result<Self> {
+        let mut validator = Validator::new_with_features(features);
```

Comment:
Similar to `wasm-compose` I think it's fine to use `WasmFeatures::all()` here to avoid needing to thread features around